cdc/metadata.cc

```diff
@@ -65,6 +65,13 @@ cdc::metadata::container_t::const_iterator cdc::metadata::gen_used_at(api::times
     return std::prev(it);
 }
 
+void cdc::metadata::clean_gens(api::timestamp_type ts) {
+    auto it = gen_used_at(ts - generation_leeway.count());
```

Comment:
.count() embeds an assumption about the units of generation_leeway.

More accurate is to subtract (generation_leeway / 1us) which converts it to microseconds no matter what the original unit was (if subtraction is actually needed).

Reply:
The type of `generation_leeway` is `api::timestamp_clock::duration`:
```cpp
extern const api::timestamp_clock::duration generation_leeway =
    std::chrono::duration_cast<api::timestamp_clock::duration>(std::chrono::seconds(5));
```
The second commit of this PR (the one adding the error injection) preserves this definition.

So, IIUC, nothing can go wrong here. I think we can assume that `api::timestamp_clock::duration` matches the units of `api::timestamp_type`.

> More accurate is to subtract (generation_leeway / 1us) which converts it to microseconds no matter what the original unit was (if subtraction is actually needed).

This solution assumes that `api::timestamp_clock::duration` uses microseconds. I think it's even more implementation-dependent.

In v2 of this PR, I didn't do any changes regarding this comment because I don't understand the problem. If I misunderstood something, could you please clarify?